api/v1alpha1/repository_types.go

```diff
@@ -28,14 +51,38 @@ type RepositorySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Repository. Edit repository_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// URL chart repository address
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
+
+	// BasicAuthSecret if the chart repository requires basic auth authentication,
+	// set the username and password to secret, with the fields user and password respectively.
+	BasicAuthSecret string `json:"basicAuthSecret,omitempty"`
+
+	// SkipTLS will not validate the repository's certificate
+	SkipTLS bool `json:"skipTLS,omitempty"`
+
+	// CAData If certification is required and the certificate is self-signed,
+	// you need to provide the server's certificate information.
+	CAData []byte `json:"caData,omitempty"`
+}
```

Comment:
I have no idea about displayname, should it in spec or annotation?
```go
DisplayName string `json:"displayName,omitempty"`
```
or
```
DisplayNameAnnotationKey = "core.kubebb.k8s.com.cn/displayName"
```

Reply:
i don't see any differences... all this should be  personal practices? Putting into annotation is good when we want all fields in `spec` be functional. When I say `functional` , I mean filed changes will 100% cause a controller reconcile

Reply:
As we only use this field for display,maybe Chinese descriptions. Therefore, I think putting it in the annotation is the best choice.

Reply:
@0xff-dev

Reply:
I agree to use annotations.